lib/src/storage/backend/rocksdb.rs

```diff
@@ -749,6 +928,11 @@ impl Reader {
                         ));
                     }
                 }
+                InnerReader::Secondary(inner) => {
+                    let db = secondary_db_handler_or_error(&inner.db)?.db;
+                    ffi_result!(rocksdb_try_catch_up_with_primary_with_status(db))?;
```

Comment:
Shouldn't we do it when we open a reader in the `snapshot` method?

Reply:
this works, but `storage.snapshot()` could then return an error. i stopped at `quads_for_pattern` which seems to not to be able to return an error and is part of the public api? so i added an `unwrap()` for now and it seems to work.